### DISCUSSION TITLE
开源社群需要什么代码贡献？ | 夜天之书

### DISCUSSION THREAD

**Discussion by giscus** [tisonkun/blog#15] [Comments]
# 开源社群需要什么代码贡献？ | 夜天之书

几天前 @Xuanwo 的一篇文章《开源运营当论迹不论心》讨论了 TDengine 代码灭虫计划活动中不尊重开源协同的规律，只是通过市场运营手段强行把开发者推进来的误区。 本文从这个例子出发，进一步举例讨论开源社群需要什么样的代码贡献。

https://tisonkun.org/2022/04/10/code-contributions/

**Comment by Endle** [comment]
给本文捉一个小虫， TDengine 代码灭虫计划 的链接是http而不是https

**Reply by tisonkun** [reply]
Fixed in https://github.com/tisonkun/tisonkun.github.io/commit/9993bf7899395ef946afcbbaf486eafad0b7693e.

**Comment by gaocegege** [comment]
https://github.com/apache/flink-kubernetes-operator Flink K8s Operator 竟然是 Java 写的。。

**Reply by tisonkun** [reply]
是啊，Flink 本身跟 Kubernetes 打交道也是用的 fabric8.io，这个其实就跟 kube-rs 写好了差不多。毕竟这个圈子里的都是 Java 代码手，有个还过得去的 Java 库写起来顺手就行了。

**Reply by gaocegege** [reply]
fabric8.io 早期特别拉胯，现在不太清楚，如果好用的话确实

**Comment by ronhuafeng** [comment]
从你文中的观点收到了很多启发。

“代码总是有不止有种方法写成” -> “代码总是有不止一种方法写成”？

**Reply by tisonkun** [reply]
Fixed by https://github.com/tisonkun/tisonkun.github.io/commit/615c72945ee0bcef6e2fcdc616a1ad32f677bb06
